src/pack-n-test.ts

```diff
@@ -27,29 +27,46 @@ const writeFileP = promisify(writeFile);
 const tmpDirP = promisify(tmp.dir) as () => Promise<string>;
 const rimrafP = promisify(rimraf);
 
-export interface BaseCodeSample {
+export interface CodeSample {
+  js?: string;
```

Comment:
I'm just wondering, would it be easier if we just did
```ts
export interface CodeSample {
  code: string;
  extension: 'js' | 'cjs' | 'esm' | 'mjs' | 'ts'; // or a string enum
  ...
}
```
that would probably simplify some of the logic below.

Reply:
While I love the idea I'd rather work on it in a follow up PR, specially given that it's going to be a semver-major breaking change for users. I'd prefer to ship this as a semver-minor addition and I'm trying to set some time to continue with some clean up work on this module during this quarter 😊

Reply:
Sounds good, agreed!